src/templates.ts

```diff
@@ -1,35 +1,58 @@
-import { loadAsync } from 'jszip'
+import JSZip, { loadAsync } from 'jszip'
 import { render } from 'mustache'
-import { RepositoryDetails, RepositoryConfiguration, TemplateInformation, Templates, OctokitInstance } from './types'
+import {
+  ExtractedContent,
+  OctokitInstance,
+  RepositoryConfiguration,
+  RepositoryDetails,
+  Template,
+  TemplateInformation,
+  Templates,
+} from './types'
 import { OctokitResponse } from '@octokit/types'
 import { Logger } from 'probot'
 
-export const extractZipContents =
-  (contents: ArrayBuffer, configuration: RepositoryConfiguration) => async (log: Logger) => {
+import { matchFile, parse } from 'codeowners-utils'
+
+const extract =
+  (loaded: JSZip, source: string) =>
+  async (log: Logger): Promise<string> => {
+    const found = loaded.file(new RegExp(source, 'i'))
+    log.debug(`Found ${found.length} file(s) matching ${source}. `)
+    const picked = found.shift()
+    if (picked) log.debug(`Using ${picked.name} for ${source}. `)
+
+    const text = (await picked?.async('text')) ?? ''
+    return text?.replace(/#{{/gm, '{{')
+  }
+
+const extractZipContents =
+  (contents: ArrayBuffer, configuration: RepositoryConfiguration) =>
+  async (log: Logger): Promise<ExtractedContent> => {
     log.debug(`Extracting ZIP contents.`)
     const loaded = await loadAsync(contents)
 
     const toProcess = Promise.all(
       configuration.files?.map(async file => {
-        const found = loaded.file(new RegExp(file.source))
-        log.debug(`Found ${found.length} file(s) matching ${file.source}. `)
-        const picked = found.shift()
-        if (picked) log.debug(`Using ${picked.name} for ${file.source}. `)
-
-        const text = (await picked?.async('text')) ?? ''
-        const contents = text?.replace(/#{{/gm, '{{')
-
+        const contents = await extract(loaded, file.source)(log)
         return {
-          path: file.destination,
+          sourcePath: file.source,
+          destinationPath: file.destination,
           contents,
         }
       }) ?? [],
     )
 
     const templates = (await toProcess).filter(it => it?.contents)
+    //TODO include in the promise all
+    const codeOwners = await extract(loaded, 'CODEOWNERS')(log)
+
     log.debug(`Extracted ${templates.length} ZIP templates.`)
 
-    return templates
+    return {
+      codeOwners: codeOwners,
+      templates: templates,
+    }
```

Comment:
When the properties are named the same as the `const`, then you can skip the colon notation 😄 

Like this:
```suggestion
    return {
      codeOwners,
      templates,
    }
```